src/aleph/http/client.clj

```diff
@@ -498,401 +957,157 @@
+
+        logger (cond
+                 (instance? LoggingHandler log-activity) log-activity
+                 (some? log-activity) (netty/activity-logger "aleph-client" log-activity)
+                 :else nil)
+
+        pipeline-builder (make-pipeline-builder
+                           (assoc opts
+                                  :ssl? ssl?
+                                  :ssl-context ssl-context
+                                  :remote-address remote-address
+                                  :raw-stream? raw-stream?
+                                  :response-buffer-size response-buffer-size
+                                  :logger logger
+                                  :pipeline-transform pipeline-transform))
+
+        ch (netty/create-client-chan
+             {:pipeline-builder    pipeline-builder
+              :bootstrap-transform bootstrap-transform
+              :remote-address      remote-address
+              :local-address       local-address
+              :transport           (netty/determine-transport transport epoll?)
+              :name-resolver       name-resolver})]
+
+    (d/chain' ch
+              (fn setup-client
+                [^Channel ch]
+                (log/debug "Channel:" ch)
+
+                ;; We know the SSL handshake must be complete because create-client wraps the
+                ;; future with maybe-ssl-handshake-future, so we can get the negotiated
+                ;; protocol, falling back to HTTP/1.1 by default.
+                (let [pipeline (.pipeline ch)
+                      protocol (if ssl?
+                                 (or (-> pipeline
+                                         ^SslHandler (.get ^Class SslHandler)
+                                         (.applicationProtocol))
+                                     ApplicationProtocolNames/HTTP_1_1)
+                                 ApplicationProtocolNames/HTTP_1_1)
+                      setup-opts (assoc opts
+                                        :authority authority
+                                        :ch ch
+                                        :keep-alive? keep-alive?
+                                        :keep-alive?' keep-alive?'
+                                        :logger logger
+                                        :non-tun-proxy? non-tun-proxy?
+                                        :pipeline pipeline
+                                        :pipeline-transform pipeline-transform
+                                        :raw-stream? raw-stream?
+                                        :remote-address remote-address
+                                        :response-buffer-size response-buffer-size
+                                        :ssl-context ssl-context
+                                        :ssl? ssl?)]
+                  (log/debug (str "HTTP protocol: " protocol))
+
+                  (let [version-specific-req-fn
```

Comment:
Call we call this something like `request-handler`, `req-handler` or `req-fn` instead?

Reply:
The problem those names are either taken, or awfully similar to existing names. 😞 

I wanted to add names to lambdas for clarification and improved stack traces, but especially on the HTTP1 side, I started running out of useful names...because what code was in which function was never designed in a principled way that lent itself to a good name.

I guess `req-handler` is technically available...but it sounds a lot like `http-req-fn` right below. What about `http-req-handler` or `http-version-req-handler`?

Reply:
How's `http-req-handler`?